kernel/model/asset_content.go

```diff
@@ -604,3 +608,98 @@ func (parser *XlsxAssetParser) Parse(absPath string) (ret *AssetParseResult) {
 	}
 	return
 }
+
+// PdfAssetParser parser factory product
+type PdfAssetParser struct {
+}
+
+// Parse will parse a PDF document using PDFium webassembly module
+func (parser *PdfAssetParser) Parse(absPath string) (ret *AssetParseResult) {
+	if !strings.HasSuffix(strings.ToLower(absPath), ".pdf") {
+		return
+	}
+
+	if !gulu.File.IsExist(absPath) {
+		return
+	}
+
+	tmp := copyTempAsset(absPath)
+	if "" == tmp {
+		return
+	}
+	defer os.RemoveAll(tmp)
+
+	f, err := os.Open(tmp)
+	if nil != err {
+		logging.LogErrorf("open [%s] failed: [%s]", tmp, err)
+		return
+	}
+	defer f.Close()
+
+	stat, err := f.Stat()
+	if nil != err {
+		logging.LogErrorf("open [%s] failed: [%s]", tmp, err)
+		return
+	}
+
+	// initialize pdfium with one worker
+	pool, err := webassembly.Init(webassembly.Config{
+		MinIdle:  1,
+		MaxIdle:  1,
+		MaxTotal: 1,
```

Comment:
As a side note: The maximum number of workers could be adjusted to number of CPU cores, but I haven't tested this with a large PDF text extraction if it would speed up processing using the WASM.